src/swarm/neo/client/mixins/RequestCore.d

```diff
@@ -375,6 +375,57 @@ public template RequestCore ( RequestType request_type_, ubyte request_code,
         assert(false);
     }
 
+    /***************************************************************************
+
+        Private helper function to handle notification of the global supported
+        codes, including the RequestSupported code.
+
+        Params:
+            status = code received from node
+            context = deserialized request context
+            remote_address = address of node which returned this status code
+
+        Returns:
+            false if `status` was a status error code and the user notified;
+            true if `status` is a RequestSupported code indicating that the
+            request is supported and that the request will start.
+
+    ***************************************************************************/
+
+    private static bool handleSupportedCodes ( SupportedStatus status,
```

Comment:
Actually, is there any reason to not deprecate `handleGlobalStatusCodes`?

Reply:
Yes, we don't have suitable replacement for the old requests which can't yet move to the new handshake mechanism.

Reply:
Mmm, isn't `handleSupportedCodes` it?

Reply:
No, because `handleSupportedCodes` can't tell the client if the actual status code was (previously) `GlobalStatusCode` or a request specific code - it returns false for all of them (it returns true only if the `RequestStarted` happens, which is never the case in the old request). In contrast, `handleGlobalStatusCodes` would return `true` if the status code was a global status code, but `false` if it was the request-specific status code.

Perhaps the deprecation path could be checking if the code is positive after casting it to `SupportedStatus` (signed byte), but that doesn't strike me as a nice solution.

Reply:
Maybe I'm missing something...

A replacement for a deprecated method doesn't have to be a _direct_ replacement. It may require changes other than changing the name of the symbol used. The deprecation just indicates that something is changing and that all users of the code will have to migrate. Which is exactly the case we have here. Or?

Reply:
The problem is that I don't see a way to fix the old, but yet supported requests, without the change in the _protocol_ (this is in contrast to our usual deprecations that can be fixed without changing the on-wire protocol). That means that either we need to make a new version of the request, or we're stuck with the silly deprecation warning for which the proto users (so, not swarm users, but the users of the concrete proto implementations) can't do anything. Now, making a new version of the request just for this change, to get rid of this particular deprecation might be fine, but I was erring on the side of not having to do this.

Reply:
(I probably didn't use `erring` in a good context here).

Reply:
Ah I see. You're right.